.github/workflows/Unit-Test.yml

```diff
@@ -20,48 +20,38 @@ jobs:
       - name: List all changed package files
         uses: tj-actions/changed-files@v44
         if: steps.changed-ruby-files.outputs.any_changed == 'true'
+      - name: Unit Tests (x86_64)
         env:
           ALL_CHANGED_FILES: ${{ steps.changed-ruby-files.outputs.all_changed_files }}
-      - name: Unit Tests (x86_64)
         run: |
-            sudo docker run --rm -t satmandu/crewbuild:m90-x86_64 sudo -i -u chronos /bin/bash -c "
+            echo "ALL_CHANGED FILES is ${ALL_CHANGED_FILES}." && \
+            export CHANGED_PACKAGES="$(echo ${ALL_CHANGED_FILES#packages/} | tr -d '.rb' | sort)" && \
+            echo "CHANGED PACKAGES is ${CHANGED_PACKAGES}." && \
```

Comment:
Add?
```
export X86_PACKAGES="$for p in ${CHANGED_PACKAGES}; do echo $(grep -l compatibility.*x86_64.* $p); done | xargs)"
```
Then if any packages exist with x86_64 architecture you can launch the container.  Otherwise, it would be pointless, correct?  Same logic could be used with arm and i686 containers as well.

Reply:
Of course, this logic would be exempt if any of the changed files affect crew itself or are outside the packages directory.

Reply:
Hmmm, we need to see what the working directory is when a github action starts...

Reply:
Actually, we need to include `all` in the filtering too.
```
export X86_64_PACKAGES="$(for p in ${CHANGED_PACKAGES}; do echo $(grep -l \"compatibility.*[all|x86_64]\" packages/$p.rb); done | xargs)"
```

Reply:
Got that already... :)

Reply:
Do we need the brackets though around all and x86_64?

Reply:
Try and find out.  It worked for me with them.

Reply:
Looks like I got it, and early escape from loading the armv7l container if the package doesn't contain any armv7l packages is in there too.